src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -564,32 +548,56 @@ public HttpClientChannelAsyncRequest(HttpClientRequestChannel channel)
                         if (_channel.State != CommunicationState.Opened)
                         {
                             // if we were aborted while getting our request or doing correlation, 
-                            // we need to abort the web request and bail
+                            // we need to abort the request and bail
                             Cleanup();
                             _channel.ThrowIfDisposedOrNotOpen();
                         }
 
-                        HttpOutput httpOutput = HttpOutput.CreateHttpOutput(_httpRequestMessage, _factory, request, _factory.IsChannelBindingSupportEnabled, _factory.GetHttpClient(), _factory._authenticationScheme, _timeoutHelper);
+                        bool suppressEntityBody = PrepareMessageHeaders(request);
 
+                        if (!suppressEntityBody)
+                        {
+                            _httpRequestMessage.Content = MessageContent.Create(_factory, request, _timeoutHelper);
+                        }
+
+                        await SendPreauthenticationHeadRequestIfNeeded();
                         bool success = false;
+
                         try
                         {
-                            _httpResponseMessageTask = await httpOutput.SendAsync(_httpRequestMessage);
-                            //this.channelBinding = httpOutput.TakeChannelBinding();
-                            await httpOutput.CloseAsync();
+                            _httpResponseMessage = await _httpClient.SendAsync(_httpRequestMessage, HttpCompletionOption.ResponseHeadersRead, _timeoutHelper.CancellationToken);
+                            _httpRequestMessage.Dispose();
```

Comment:
It would be useful to comment why success is assumed here if no exception is thrown.  Is there nothing in the status code or headers that we care about?

Also, shouldn't we set _httpResponseMessage to null after the Dispose -- just to avoid holding onto disposed objects?